app/services/cycle_timetable.rb

```diff
@@ -162,7 +168,8 @@ def self.current_cycle?(application_form)
   end
 
   def self.can_add_course_choice?(application_form)
-    return true if Time.zone.now.to_date >= find_reopens && !application_form.must_be_carried_over?
+    return false if application_form.must_be_carried_over?
+    return true if Time.zone.now.to_date >= find_reopens
```

Comment:
Will this ever be true? `find_reopens` is now by definition a date in the _next_ cycle

Reply:
@duncanjbrown @davidgisbey - we commented out line 172 (`find_reopens`) and the entire test suite has still passed.  I'm not entirely sure what the implications are regarding business rules. Can you please advise?

Reply:
Good excuse to add a spec for this method!

Reply:
Actually there is a spec it passes because another one of the condition is true. You are right Duncan - it is now impossible for line 172  to ever resolve to 'true' as it now always looks ahead. I'm wondering if this is has implications or if that line can simply be removed?

Reply:
Interestingly, there are specs for this (e.g. https://github.com/DFE-Digital/apply-for-teacher-training/pull/5106/files#diff-fad1017966158907e50ba7ac1d694af33bc81613fb3d7771a667d78d806615c4R168) however you are 100% right Duncan. Line 172 will now always resolve to 'true' as we are now always looking ahead to when Find reopens. The reason why the specs pass is because of another condition below resolving to true.

I'm not sure if for now we can simply delete that line or if this has implications that we have not thought of.

Reply:
David has suggested some tweaks to the '.can_add_course_choice?' which deals with the issues above and IMO greatly improves readability. Thanks @davidgisbey 
https://github.com/DFE-Digital/apply-for-teacher-training/pull/5106/commits/01d5d2774b0343481e491b5efdafa9df51cd1ced